envoy/stream_info/filter_state.h

```diff
@@ -102,21 +104,17 @@ class FilterState {
      * This method can be used to get an unstructured serialization result.
      */
     virtual absl::optional<std::string> serializeAsString() const { return absl::nullopt; }
-  };
-
-  /**
-   * Generic reflection support for the filter state objects.
-   */
-  class ObjectReflection {
-  public:
-    virtual ~ObjectReflection() = default;
 
-    using FieldType = absl::variant<absl::monostate, absl::string_view, int64_t>;
+    /**
+     * @return bool true if the object supports field access. False if the object does not support
+     * field access. Default implementation returns false.
+     */
+    virtual bool hasFieldSupport() const { return false; }
```

Comment:
This is only used to support the case where we need to provide different behavior for same object and same calling. (Now, only the expr need this. If we can change the expr in future, we can remove this method.)